renku/api/models/run.py

```diff
@@ -58,17 +59,30 @@
 )
 from renku.core.plugins.provider import RENKU_ENV_PREFIX
 
+name_pattern = re.compile("[a-zA-Z0-9-_]+")
+
 
 class _PathBase(PathLike):
     @ensure_project_context
-    def __init__(self, path, project):
-        self._path = Path(path)
+    def __init__(self, name: str, path: Union[str, Path], project):
```

Comment:
It should be possible to automatically generate names for Input/Output. I believe it has a better UX.

Reply:
We need to update our discourse post about the Renku Python API if we decide to force users to provide a name.